### PR TITLE
mock student generator bug fix

### LINKED ISSUE #232: Fix mock student generator
### Issue
- The `mock_student_generator` does not assign the correct number of friends and enemies
- The enemy generation isn't safe either

### PR DESCRIPTION
**Pull request by justino599**
Closes #232 
Description in issue. Reasoning for fix is that since specifying 2 enemies creates two enemy relationships on each student, specifying 2 friends should also do the same. The old behaviour is that each student would be a part of a clique of 2, rather than have 2 friends

### CODE REVIEW

**Review by Opeyem1a — CHANGES_REQUESTED** [review]
I think I need to see some tests here to be convinced you fixed it. In general this function is kinda sus and was stolen from an old time. If we need it to work we should be sure it works fully yknow

File: tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py [hunk 1]
```diff
@@ -105,7 +175,7 @@ def test_create_mock_students__student_has_num_values_for_attribute_within_range
     def test_create_mock_students__students_have_correct_friend_and_enemy_count(self):
         for i in [1, 5, 10]:
             students = create_mock_students(
-                number_of_students=i + 1,
+                number_of_students=i * 2 + 1,
```
Comment (Opeyem1a) [inline tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:178]: Q: why was this change needed?

File: tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py [hunk 2]
```diff
@@ -21,6 +21,76 @@ def test_get__returns_student_objects(self):
...
+            num_friends = len(
+                [
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.FRIEND
+                ]
+            )
+            num_enemies = len(
+                [
+                    1
```
Comment (Opeyem1a) [inline tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:43]: nit; make this 2, I think the way it worked before could have still passed this test if you use 1

File: tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py [hunk 3]
```diff
@@ -21,6 +21,76 @@ def test_get__returns_student_objects(self):
         for student in students:
             self.assertIsInstance(student, Student)
 
+    def test_get__each_student_has_correct_num_friends_and_enemies(self):
```
Comment (Opeyem1a) [inline tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:24]: these tests actually should go in the Helpers test class, since the function that generates students isn't in MockStudentProvider, it is just used there

File: tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py [hunk 4]
```diff
@@ -21,6 +21,76 @@ def test_get__returns_student_objects(self):
...
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.ENEMY
+                ]
+            )
+
+            self.assertEqual(3, num_friends)
+            self.assertEqual(1, num_enemies)
+
+    def test_get__class_size_too_small(self):
```
Comment (Opeyem1a) [inline tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:82]: I'm being a bit greedy here, but you've added a few new things to care about that should be tested (I believe):
- a student's friend cannot also be their enemy, if there are enough people
- if cluster is used, we should check the cluster is a cluster

**Review by Opeyem1a — CHANGES_REQUESTED** [review]
I'm bullying you

File: tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py (hunk 1 above)
Comment (justino599) [reply tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:178]: Cause otherwise in the `i=1` case there aren't enough students in the class to satisfy the friends and enemies req

Comment (Opeyem1a) [reply tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py:178]: Ah gotcha

**Review by Opeyem1a — APPROVED** [review]
🐐
